app/src/main/java/com/android/ubclaunchpad/driver/session/SessionInfoActivity.java

```diff
@@ -215,14 +180,148 @@ public void onCancelled(DatabaseError databaseError) {
+            @Override
+            public void onChildAdded(DataSnapshot dataSnapshot, String s) {
+                addUserToAdapter(dataSnapshot, adapter, true);
+            }
+
+            @Override
+            public void onChildChanged(DataSnapshot dataSnapshot, String s) {
+            }
+
+            @Override
+            public void onChildRemoved(DataSnapshot dataSnapshot) {
+                removeUserFromAdapter(dataSnapshot, adapter, true);
+            }
+
+            @Override
+            public void onChildMoved(DataSnapshot dataSnapshot, String s) {
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+            }
+        };
+
+        passengersListener = new ChildEventListener() {
+            @Override
+            public void onChildAdded(DataSnapshot dataSnapshot, String s) {
+                addUserToAdapter(dataSnapshot, adapter, false);
+            }
+
+            @Override
+            public void onChildChanged(DataSnapshot dataSnapshot, String s) {
+
+            }
+
+            @Override
+            public void onChildRemoved(DataSnapshot dataSnapshot) {
+                removeUserFromAdapter(dataSnapshot, adapter, false);
+            }
+
+            @Override
+            public void onChildMoved(DataSnapshot dataSnapshot, String s) {
+
+
+            }
+
+            @Override
+            public void onCancelled(DatabaseError databaseError) {
+
+            }
+        };
+
+
+
+        driverPassengersListener = new ChildEventListener() {
+
+            @Override
+            public void onChildAdded(final DataSnapshot sessionSnapshot, String s) {
+
+                DatabaseReference usersReference = FirebaseUtils.getDatabase().child(StringUtils.FirebaseUserEndpoint);
+                usersReference.addListenerForSingleValueEvent(new ValueEventListener() {
+                    @Override
+                    public void onDataChange(DataSnapshot usersSnapshot) {
+
+                        // If algorithm finished
+                        if (sessionSnapshot.getKey().equals(StringUtils.FirebaseSessionDriverPassengers)) {
+
+                            User currentUser = usersSnapshot.child(UID).getValue(User.class);
+                            DataSnapshot driverPassengersSnapshot = sessionSnapshot.child(StringUtils.FirebaseSessionDriverPassengers);
+                            Bundle data = new Bundle();
+                            Message message = new Message();
+                            message.setData(data);
+                            data.putString(StringUtils.FirebaseSessionName, sessionName);
+                            if (currentUser.getIsDriver()) {
+
+                                // No need to search for anything, just start DriverPassengersActivity
+                                data.putString("driverUid", UID);
+                                runOnUiThread(startActivityTask(data));
```

Comment:
Nope, it only runs when driverPassengers map appears on the session object in firebase, which happens only after you hit Go button.